### PR TITLE
Upgrade PHP ecoCode plugin to 1.4.2

### PR DESCRIPTION
**Pull request by dedece35**
Hi,

ecoCode PHP plugin 1.4.2 has been released.

Changes from last SonarQube marketplace integration (1.3.0), please check following changelogs :

from 1.3.0 to 1.4.0 : https://github.com/green-code-initiative/ecoCode-php/releases/tag/1.4.0
from 1.4.0 to 1.4.1 : https://github.com/green-code-initiative/ecoCode-php/releases/tag/1.4.1
from 1.4.1 to 1.4.2 : https://github.com/green-code-initiative/ecoCode-php/releases/tag/1.4.2

The corresponding SonarCloud project: https://sonarcloud.io/project/overview?id=green-code-initiative_ecoCode-php

Please update the market place.

Thank you

Regards

### CODE REVIEW

**Comment by dedece35** [comment]
Hi @ganncamp,
any news for his PR ? 😜

thank you.
regards,
David

File: ecocodephp.properties [hunk 1]
```diff
@@ -13,8 +13,14 @@ defaults.mavenArtifactId=ecocode-php-plugin
 1.2.1.downloadUrl=https://github.com/green-code-initiative/ecoCode/releases/download/1.2.1/ecocode-php-plugin-1.2.1.jar
 1.2.1.changelogUrl=https://github.com/green-code-initiative/ecoCode/releases/tag/1.2.1
 
-1.3.0.description=Updates rules tags�; adds rule (EC3)�; upgrades librairies to SonarQube 10.0.0�; clean-up plugins and dependencies
-1.3.0.sqVersions=[9.9,LATEST]
+1.3.0.description=Updates rules tags; adds rule (EC3); upgrades librairies to SonarQube 10.0.0; clean-up plugins and dependencies
+1.3.0.sqVersions=[9.9,10.1]
```
Comment (ganncamp) [inline ecocodephp.properties:17]: This range needs to end with the current version so that your current users on SQ 10.3 will be prompted to upgrade.

**Review by ganncamp — CHANGES_REQUESTED** [review]

File: ecocodephp.properties (hunk 1 above)
Comment (dedece35) [reply ecocodephp.properties:17]: ok. thank you. I'm going to do this

Comment (dedece35) [reply ecocodephp.properties]: done

**Comment by dedece35** [comment]
@ganncamp with recent correction of "LATEST" value on all versions in order to enable upgrade old plugins, I have an error on build step with "Only the latest release of plugin ecocodephp may depend on LATEST SonarQube" message.
I undertsand that only the last version can have "LATEST" value, isn't it ?
(this problem is the same for the 2 others PR for Python and Java plugin)

File: ecocodephp.properties [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+homepageUrl=https://github.com/green-code-initiative/ecoCode-php
+publicVersions=1.4.2
+archivedVersions=1.2.1,1.3.0
 
 defaults.mavenGroupId=io.ecocode
 defaults.mavenArtifactId=ecocode-php-plugin
 
 1.2.1.description=Initial Marketplace publication of plugin for SonarQube 9.9.+ LTS
-1.2.1.sqVersions=[9.9,10.1]
+1.2.1.sqVersions=[9.9,LATEST]
```
Comment (ganncamp) [inline ecocodephp.properties:11]: Please see my comments on the Java plugin

File: ecocodephp.properties [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
 1.2.1.date=2023-04-14
 1.2.1.downloadUrl=https://github.com/green-code-initiative/ecoCode/releases/download/1.2.1/ecocode-php-plugin-1.2.1.jar
 1.2.1.changelogUrl=https://github.com/green-code-initiative/ecoCode/releases/tag/1.2.1
 
-1.3.0.description=Updates rules tags�; adds rule (EC3)�; upgrades librairies to SonarQube 10.0.0�; clean-up plugins and dependencies
+1.3.0.description=Updates rules tags; adds rule (EC3); upgrades librairies to SonarQube 10.0.0; clean-up plugins and dependencies
 1.3.0.sqVersions=[9.9,LATEST]
```
Comment (ganncamp) [inline ecocodephp.properties:17]: End this range with `10.3`

**Review by ganncamp — CHANGES_REQUESTED** [review]

File: ecocodephp.properties (hunk 3 above)
Comment (dedece35) [reply ecocodephp.properties]: all is ok now.

File: ecocodephp.properties (hunk 1 above)
Comment (ganncamp) [inline ecocodephp.properties:17]: Oops. This one really does need to end with `10.3`
The version being archived - the one that previously ended with `LATEST` needs to be updated to the current version.

**Review by ganncamp — CHANGES_REQUESTED** [review]

**Review by ganncamp — APPROVED** [review]

File: ecocodephp.properties (hunk 1 above)
Comment (dedece35) [reply ecocodephp.properties]: sorry, correction done.

**Pull request merged**